runtime/parachains/src/hrmp.rs

```diff
@@ -534,31 +606,49 @@ impl<T: Config> Pallet<T> {
 	pub(crate) fn initializer_on_new_session(
 		notification: &initializer::SessionChangeNotification<T::BlockNumber>,
 		outgoing_paras: &[ParaId],
-	) {
-		Self::perform_outgoing_para_cleanup(&notification.prev_config, outgoing_paras);
+	) -> Weight {
+		let w1 = Self::perform_outgoing_para_cleanup(&notification.prev_config, outgoing_paras);
 		Self::process_hrmp_open_channel_requests(&notification.prev_config);
 		Self::process_hrmp_close_channel_requests();
+		w1.saturating_add(<T as Config>::WeightInfo::force_process_hrmp_open(
+			outgoing_paras.len() as u32
+		))
+		.saturating_add(<T as Config>::WeightInfo::force_process_hrmp_close(
+			outgoing_paras.len() as u32
+		))
```

Comment:
I find it a little bit weird how it reads here: we use the weight data that was meant to be used (judging by the name) by `force_*` dispatchables.

Reply:
True, we're reusing the weights of the force dispatchables here since they are almost 100% the same and can be reused. Maybe this extended snipped clearifies my intention further. 

```
// is not weighed anywhere else, therefore we return its weight manually. 
Self::perform_outgoing_para_cleanup(&notification.prev_config, outgoing_paras);
// already weighed as a part of dispatchable, so we reuse it.
Self::process_hrmp_open_channel_requests(&notification.prev_config);
Self::process_hrmp_close_channel_requests();
```

Reply:
I think it's just the naming implies that the `force_` functions are central and these internal functions are secondary, whereas it is the other way around: it is just so happens that the `force_` functions are the same as those internal functions.